### ISSUE TITLE
Hourlies subtitles popping at the wrong time

### ISSUE THREAD

**Issue by Tamachan12** [KC3Kai/KC3Kai#1243]
Sometimes hourlies do pop on wrong time, mostly it triggers when you go back from any screen to your room.

http://imgur.com/p4ey0ds

**Comment by dragonjet** [comment]
hm in that case it may seem that the game client pre-loads the sounds even before the top of the hour. currently we just detect on when the sounds was loaded - in the mindset that it is immediately played after being loaded. If not, we may time the hourlies ourselves.

but i haven't checked the actual game - so these are just things that came to my mind

**Comment by Tamachan12** [comment]
Well the hourly played normally, but no subt, since it jumbled with maruyu iddle line im going to check it again to see if i can reproduce it.

**Comment by mikenzb** [comment]
Yup happened here as well, looks like the game will preload the mp3 file before the hour and when it plays it doesnt make a network request again for the file.
![capture](https://cloud.githubusercontent.com/assets/5982780/14129672/9d851746-f688-11e5-904f-4b30d7f04acb.PNG)

However interestly enough there was some request to DMM at 14:59  (which I think is unrelated)
![capture2](https://cloud.githubusercontent.com/assets/5982780/14129688/d196775a-f688-11e5-804d-b246785e557d.PNG)

**Comment by dragonjet** [comment]
seems we need to do the hourly timings ourselves

**Comment by Tamachan12** [comment]
Hmm the pre-load seems so randomic, got it at 59 now...

**Comment by ShadowfoxEX** [comment]
I'm pretty sure it's not exclusive to hourly. I'm pretty sure white day lines are playing and I'm hearing the word chocolate, but I'm not seeing it translated properly

**Comment by bcc32** [comment]
A possible option to explore is detecting audio playback from the browser and assuming that whatever line is being played is the same as the file that was most recently loaded.

https://developer.chrome.com/extensions/tabs#property-Tab-audible

**Comment by dragonjet** [comment]
@ShadowfoxEX the quotes shown vs voice played is a different issue, and is tracked in a spreadsheet. This issue is about the _timings_ of the hourly notifs.

@bcc32 I've considered that before, but if their bgm is enabled, it's always audible

**Comment by Tamachan12** [comment]
Seems the Seasonal Lines are jumbling together with the regular lines as well look at my wai.. Ooyodo case.
![tama 2016-04-03 07-32-50 76](https://cloud.githubusercontent.com/assets/14944570/14229521/2fce7306-f970-11e5-85de-88e52a021648.jpg)

**Comment by Tamachan12** [comment]
Furutaka Equipment lines are swapped as well.

**Comment by dragonjet** [comment]
@Tamachan12 
https://github.com/KC3Kai/KC3Kai/issues/751#issuecomment-202033407

**Comment by dragonjet** [comment]
idea came up when looking for a fix for something else:

when voice lines are loaded, determine number of millis until the next hour, set a timer and fire the subtitle event after timer

**Comment by mikenzb** [comment]
A workaround has been temporally implemented, will close for now but there is still room for improvement.
